File: backend/app/api/routes/system.py
from pathlib import Path

from fastapi import APIRouter

router = APIRouter()


@router.get("/health")
def health() -> dict:
    return {"status": "ok"}


@router.get("/layout")
def layout() -> dict:
    root = Path(__file__).resolve().parents[4]
    engine_root = root / "engine"
    return {
        "project_root": str(root),
        "engine_exists": engine_root.exists(),
        "backend_exists": (root / "backend").exists(),
        "frontend_exists": (root / "frontend").exists(),
        "engine_dirs": {
            "app": (engine_root / "app").exists(),
            "core": (engine_root / "core").exists(),
            "configs": (engine_root / "configs").exists(),
            "datasets": (engine_root / "datasets").exists(),
            "outputs": (engine_root / "outputs").exists(),
            "third_party": (engine_root / "third_party").exists(),
        },
    }

File: frontend/src/src/api/system.ts
import { 请求 } from './client'
import type { 健康检查响应, 布局检查响应 } from '../types/settings'

export function 获取系统健康() {
  return 请求<健康检查响应>('/system/health')
}

export function 获取系统布局() {
  return 请求<布局检查响应>('/system/layout')
}
